src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -348,7 +349,7 @@ fn vec_slice_metadata(
     let (pointer_size, pointer_align) = cx.size_and_align_of(data_ptr_type);
     let (usize_size, usize_align) = cx.size_and_align_of(cx.tcx.types.usize);
 
-    let member_descriptions = [
+    let member_descriptions = vec![
```

Comment:
Yes, because it allows to re-use the `String` allocations when constructing the `CString` for the `name` field down the line. It should be a win overall since all the other `MemberDescription` can't get around allocating anyway.

Reply:
If you want to go crazy, the LLVM C++ method we're using is called through a C wrapper that we define ourselves and uses a StringRef, which is a pointer/length pair and doesn't actually need a terminating nul byte. So you could get theoretically completely avoid CStr for the name (and possibly other places as well). Of course, if / when the DebugInfo stuff gets an official C API, we'd likely be back at using plain C strings.

Reply:
> If you want to go crazy

I don't `:P`